packages/flutter_test/lib/src/matchers.dart

```diff
@@ -1471,23 +1488,20 @@ class _MatchesGoldenFile extends AsyncMatcher {
   final Uri key;
 
   @override
-  Future<String> matchAsync(covariant Finder finder) async {
-    final Iterable<Element> elements = finder.evaluate();
-    if (elements.isEmpty) {
-      return 'could not be rendered because no widget was found';
-    } else if (elements.length > 1) {
-      return 'matched too many widgets';
-    }
-    final Element element = elements.single;
-
-    RenderObject renderObject = element.renderObject;
-    while (!renderObject.isRepaintBoundary) {
-      renderObject = renderObject.parent;
-      assert(renderObject != null);
+  Future<String> matchAsync(dynamic item) async {
+    Future<ui.Image> imageFuture;
+    if (item is FutureOr<ui.Image>) {
```

Comment:
similar here, we should just check against Future, or check against Image.

Reply:
what is the concern with using `FutureOr` in an is check?
```dart
if (item is FutureOr<ui.Image>)
```
it is more terse and may give better type promotion than 
```dart
if (item is Future<ui.Image> || item is ui.Image)
```

Reply:
we're trying to avoid anyone seeing the term `FutureOr` anywhere in the codebase, on the basis that it's confusing because it's a weird magical corner of Dart that'll eventually go away when we have union types.